Handle Slackware initrd name initrd.gz, file systemq f2fs and reiserfs and
include Slint among the Linux distributions.
Didier Spaier
diff -ru os-prober-1.77/linux-boot-probes/mounted/common/90fallback os-prober-1.77.new/linux-boot-probes/mounted/common/90fallback
--- os-prober-1.77/linux-boot-probes/mounted/common/90fallback	2018-08-10 21:23:18.000000000 +0200
+++ os-prober-1.77.new/linux-boot-probes/mounted/common/90fallback	2020-09-06 19:24:46.876019792 +0200
@@ -34,8 +34,11 @@
 			initrdname3=$(echo "$kernfile" | sed "s/vmlinu[zx]/initramfs\*/" | sed 's/$/.img/')
 			# And Gentoo's also
 			initrdname4=$(echo "$kernfile" | sed "s/kernel/initramfs\*/")
+			# Slackware names ist initrd initrd.gz, regardless the kernels for
+			# which it is built
+			inirdname5="initrd.gz"
 			foundinitrd=0
-			for initrd in $(eval ls "$initrdname" "$initrdname1" "$initrdname2" "$initrdname3" "$initrdname4" 2>/dev/null); do
+			for initrd in $(eval ls "$initrdname" "$initrdname1" "$initrdname2" "$initrdname3" "$initrdname4" "$initrdname5" 2>/dev/null); do
 				if [ "$initrd" != "$kernfile" ] && [ -f "$initrd" ] && [ ! -L "$initrd" ]; then
 					initrd=$(echo "$initrd" | sed "s!^$mpoint!!")
 					result "$partition:$kernbootpart::$kernbasefile:$initrd:root=$mappedpartition"
diff -ru os-prober-1.77/os-probes/init/common/10filesystems os-prober-1.77.new/os-probes/init/common/10filesystems
--- os-prober-1.77/os-probes/init/common/10filesystems	2018-08-10 21:23:18.000000000 +0200
+++ os-prober-1.77.new/os-probes/init/common/10filesystems	2020-09-06 19:41:35.777048655 +0200
@@ -2,7 +2,7 @@
 # Make sure filesystems are available.
 set +e	# ignore errors from modprobe
 
-FILESYSTEMS='ext2 ext3 ext4 xfs jfs msdos vfat ntfs minix hfs hfsplus qnx4 ufs btrfs'
+FILESYSTEMS='ext2 ext3 ext4 f2fs xfs jfs msdos vfat ntfs minix hfs hfsplus qnx4 ufs btrfs reiserfs'
 # fuse is needed to make grub-mount work.
 FILESYSTEMS="$FILESYSTEMS fuse"
 # The Ubuntu kernel udebs put a number of filesystem modules in
diff -ru os-prober-1.77/os-probes/mounted/common/90linux-distro os-prober-1.77.new/os-probes/mounted/common/90linux-distro
--- os-prober-1.77/os-probes/mounted/common/90linux-distro	2018-08-10 21:23:18.000000000 +0200
+++ os-prober-1.77.new/os-probes/mounted/common/90linux-distro	2020-09-06 11:10:36.486417137 +0200
@@ -110,6 +110,9 @@
 	elif [ -e "$dir/etc/kanotix-version" ]; then
 		short="Kanotix"
 		long="$(cat "$dir/etc/kanotix-version")"
+	elif [ -e "$dir/etc/slint-version" ]; then
+		short="Slint"
+		long="$(printf "Slint Linux (%s)\n" "$(cat "$dir/etc/slint-version")")"
 	elif [ -e "$dir/etc/slackware-version" ]; then
 		short="Slackware"
 		long="$(printf "Slackware Linux (%s)\n" "$(cat "$dir/etc/slackware-version")")"
